analyzers/src/SonarAnalyzer.CSharp/Rules/UnnecessaryUsings.cs

```diff
@@ -74,10 +81,14 @@ private static void CheckUnnecessaryUsings(SonarSyntaxNodeReportingContext conte
             foreach (var usingDirective in usingDirectives)
             {
                 // This will create some FNs but will kill noise from FPs.
-                // For more info see issue: https://github.com/SonarSource/sonar-dotnet/issues/5946
-                if (usingDirective.GetFirstToken().IsKind(SyntaxKind.GlobalKeyword))
+                // For more info see issues:
+                //  - https://github.com/SonarSource/sonar-dotnet/issues/5946
+                //  - https://github.com/SonarSource/sonar-dotnet/issues/7959
+                if (usingDirective.GetFirstToken().IsKind(SyntaxKind.GlobalKeyword)
+                    || (GeneratedCodeRecognizer.IsRazorGeneratedFile(usingDirective.SyntaxTree)
+                        && IgnoredRazorFiles.Contains(Path.GetFileName(usingDirective.GetLocation().GetMappedLineSpan().Path).ToUpperInvariant())))
```

Comment:
I think we should use a HashSet configured with a comparer. See: https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1.-ctor?view=net-7.0#system-collections-generic-hashset-1-ctor(system-collections-generic-iequalitycomparer((-0)))

This is prone to errors and also not optimal to do all the string transformations.